packages/mcdoc/src/runtime/checker/index.ts

```diff
@@ -498,7 +500,8 @@ function checkShallowly<T>(
 			if (
 				typeDef.valueRange
 				&& simplifiedInferred.kind === 'literal'
-				&& typeof simplifiedInferred.value.value === 'number'
+				&& simplifiedInferred.value.kind !== 'string'
+				&& simplifiedInferred.value.kind !== 'boolean'
```

Comment:
There are probably a few other places, maybe in attributes, where this pattern was used

Reply:
Pretty sure that's untrue. This one I cought because it caused a build error, so I don't see why this wouldn't happen in other places.